core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +88,45 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v30_dict_to_v31_dict(cls, draft_change_list):
+        """Converts draft change list from state version 30 to 31. State
+        Version 31 adds the duration_secs float for the Voiceover
+        section of state.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS):
+                # Get the language code to access the language code correctly.
+                new_changes = change.new_value['voiceovers_mapping']
+                # Initialize the value to migrate draft state to v31.
+                # Use the dynamic language code like 'en' to access the data.
+                # Using 0.1 as a temporary value because js uses 0.0 as 0. The
+                # reason is probably 0 == 0.0 and doesn't need to be
+                # represented as 0.0. Fun fact some processor(s) represent
+                # 0 using the same reasoning for both floats and ints.
+                for state in new_changes.keys():
+                    for language_code in new_changes[state].keys():
+                        new_changes[state][language_code]['duration_secs'] = 0.1
```

Comment:
The original plan is to have a one-off job that will populated with correct value. I'm open to combining the one-off job into this PR if needed.

Reply:
I think that sounds fine. Is it straightforward to write it? That way, we can probably test this end-to-end before the PR goes in.

If not straightforward, though, I guess we can do it in two stages.

@DubeySandeep any thoughts/recommendations from your end?

Reply:
As per discussion with @mzaman07 we are going to have the one-off job in the same PR.

Reason:
- Just to avoid having a few audios with the time data and others have false time information (01.s) waiting for one-off jobs to update them which can only happen after 30 days i.e, in the next release (if somehow the one-off job PR got delayed in merge).
- This will help us to run both the jobs in the backup server for testing in the same deployment.

Reply:
This is in progress.